lisa/schema.py

```diff
@@ -1257,7 +1257,7 @@ def __post_init__(self, *args: Any, **kwargs: Any) -> None:
         add_secret(self.admin_username, PATTERN_HEADTAIL)
         add_secret(self.admin_password)
 
-        if self.type != constants.PLATFORM_READY:
+        if self.type not in (constants.PLATFORM_READY, constants.PLATFORM_AZURE):
```

Comment:
We can remove this check.

Reply:
generating key pairs is not suitable for ready platform + when user specified the rg not from lisa run?

Reply:
updated